scipy/stats/morestats.py

```diff
@@ -2199,31 +2258,15 @@ def ansari(x, y, alternative='two-sided'):
     if repeats and (m < 55 or n < 55):
         warnings.warn("Ties preclude use of exact statistic.")
     if exact:
-        astart, a1, ifault = statlib.gscale(n, m)
-        ind = AB - astart
-        total = np.sum(a1, axis=0)
-        if ind < len(a1)/2.0:
-            cind = int(ceil(ind))
-            if cind != ind:
-                cind = cind - 1
-            if alternative == 'two-sided':
-                pval = 2.0 * np.sum(a1[:cind+1], axis=0) / total
-            elif alternative == 'greater':
-                # see [3]_
-                pval = np.sum(a1[:cind+1], axis=0) / total
-            else:
-                # see [3]_
-                pval = 1.0 - np.sum(a1[:cind], axis=0) / total
+        if alternative == 'two-sided':
+            pval = 2.0 * np.minimum(_abw_state.cdf(AB, n, m),
+                                    _abw_state.sf(AB, n, m))
```

Comment:
Do you agree this is the same? I did this because I didn't want to me with the rounding of the test statistic here. We could calculate the mean analytically and check whether it is to the left or right, but if this is correct, it seems a little simpler.

Reply:
> Do you agree this is the same? I did this because I didn't want to deal with the rounding of the test statistic here. We could calculate the median analytically and check whether the statistic value is to the left or right (and use SF or CDF as appropriate), but if this is correct, it seems a little simpler.

Yes, I agree this is the same. It does seem a lot simpler. If you think the extra complexity is not worth it, I am very happy with how this looks right now.